pkg/controller/util/node/controller_utils.go

```diff
@@ -99,7 +102,14 @@ func DeletePods(ctx context.Context, kubeClient clientset.Interface, pods []*v1.
 // pod status, updates it in the apiserver, and returns an error if it
 // encounters one.
 func SetPodTerminationReason(ctx context.Context, kubeClient clientset.Interface, pod *v1.Pod, nodeName string) (*v1.Pod, error) {
-	if pod.Status.Reason == nodepkg.NodeUnreachablePodReason {
+	var updatedCondition bool = false
+	if feature.DefaultFeatureGate.Enabled(features.JobBackoffPolicy) {
+		updatedCondition = utilpod.UpdatePodCondition(&pod.Status, &v1.PodCondition{
+			Type:   v1.EvictedFromUnhealtyNode,
```

Comment:
It makes sense to use the same pod condition in these three situations:
- garbage collected from a node tainted as out of service
- garbage collected from unreachable node
- deleted from unreachable node by node lifecycle controller

I suggest "DeletedFromUnsuableNode" as the name. Unreachable might be misleading cause it might be
"reachable", but tainted out of service.

Reply:
sgtm